.github/workflows/openssl_versions.yml

```diff
@@ -0,0 +1,156 @@
+---
+name: OpenSSL Versions
+
+on:
+  pull_request:
+    branches: ["main"]
+  push:
+    branches: ["main"]
+
+jobs:
+  build:
+    name: Versions
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: ossl30
+            build: true
+            version: "3.0"
+            container: fedora:latest
+          - name: ossl32
+            build: true
+            version: "3.2"
+            container: fedora:latest
+          - name: ossl35
+            build: true
+            version: "3.5"
+            container: fedora:latest
+          - name: debian
+            build: false
+            version: "3.5"
+            container: debian:testing
+          - name: centos10
+            build: false
+            version: "3.5"
+            container: quay.io/centos/centos:stream10
+    container: ${{ matrix.container }}
+    steps:
+      - name: Get Date for DNF cache entry
+        id: get-date
+        run: |
+          echo "date=$(/bin/date -u "+%Y%V")" >> $GITHUB_OUTPUT
+        shell: bash
+
+      - name: Restore DNF cache
+        uses: actions/cache/restore@v4
+        with:
+          path: |
+            /var/cache/libdnf5
+          key: ${{ runner.os }}-dnf-${{ steps.get-date.outputs.date }}
```

Comment:
This should be skipped on debian. But for some reason, it does not restore the cache on fedora either.
Also we should have separate cache for fedora and centos10.

Reply:
ok let me check